### PR TITLE
[bitnami/thanos] Added remote-write in service and ingress also make srv record discovery possible.

### PR DESCRIPTION
**Pull request by Cellebyte**
**Description of the change**

Added support for the remote-write on ingress.

Fixed a minor bug.

Finally fixed the dnsDiscovery mechanism because the thanos queriers were not able to correctly discover the other components.

```yaml
additionalHeadless: true
```
is now default

**Benefits**

Remote Write can be exposed via an ingress.

**Possible drawbacks**

It's not possible to disable the functionality when the ingress gets enabled.

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

**Review by alemorcuq — CHANGES_REQUESTED** [review]
Thank you for yet another contribution, @Cellebyte ! Could you please rebase? The remote write port was already added to the service by another user in #5430, but the Ingress is still missing. The version increase should also be a minor instead of a patch, since you are adding new functionality.

**Comment by Cellebyte** [comment]
@alemorcuq please not merge.
Iam currently trying the following to solve.
When you want to use thanos dns discovery you need a service with "clusterIP: None", but then you are not anymore able to use the ingress.
Should we cover such a use-case because it only affects the `grpc` port of the container.

**Comment by Cellebyte** [comment]
@alemorcuq fixed some bugs coming along the way.
Is the semver version still fine?
Best Regards.

File: bitnami/thanos/Chart.yaml [hunk 1]
```diff
@@ -28,4 +28,4 @@ name: thanos
 sources:
   - https://github.com/bitnami/bitnami-docker-thanos
   - https://thanos.io
-version: 3.8.4
+version: 3.8.5
```
Comment (alemorcuq) [inline bitnami/thanos/Chart.yaml:31]:
```suggestion
version: 3.9.0
```

File: bitnami/thanos/templates/query/deployment.yaml [hunk 2]
```diff
@@ -80,10 +80,13 @@ spec:
             - --store=dnssrv+_grpc._tcp.{{- include "common.tplvalues.render" ( dict "value" $query.dnsDiscovery.sidecarsService "context" $) -}}.{{- include "common.tplvalues.render"  ( dict "value" $query.dnsDiscovery.sidecarsNamespace "context" $) -}}.svc.{{ .Values.clusterDomain }}
             {{- end }}
             {{- if and .Values.storegateway.enabled $query.dnsDiscovery.enabled }}
-            - --store=dnssrv+_grpc._tcp.{{ include "common.names.fullname" . }}-storegateway.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}
+            - --store=dnssrv+_grpc._tcp.{{ include "common.names.fullname" . }}-storegateway-headless.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}
```
Comment (alemorcuq) [inline bitnami/thanos/templates/query/deployment.yaml:83]: The headless service can be disabled. What happens if the headless service is disabled?

Comment (Cellebyte) [reply bitnami/thanos/templates/query/deployment.yaml:83]: hmm we could add it as a flag.
headless service just makes more sense.

Comment (Cellebyte) [reply bitnami/thanos/templates/query/deployment.yaml]: like that?

**Comment by romainmeunier** [comment]
Can we add a service discovery for the **receive** headless service in this PR ?

**Comment by Cellebyte** [comment]
> Can we add a dns service discovery entry for the **receive** headless service in this PR ?

I just added all as it is before and i could change the default so that it behaves like before and is opt-in

File: bitnami/thanos/templates/query/deployment.yaml [hunk 3]
```diff
@@ -80,10 +80,13 @@ spec:
...
-            - --store=dnssrv+_grpc._tcp.{{ include "common.names.fullname" . }}-storegateway.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}
+            - --store=dnssrv+_grpc._tcp.{{ include "common.names.fullname" . }}-storegateway{{ if .Values.storegateway.service.additionalHeadless }}-headless{{ end }}.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}
             {{- end }}
             {{- if and .Values.ruler.enabled $query.dnsDiscovery.enabled }}
-            - --store=dnssrv+_grpc._tcp.{{ include "common.names.fullname" . }}-ruler.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}
+            - --store=dnssrv+_grpc._tcp.{{ include "common.names.fullname" . }}-ruler{{ if .Values.ruler.service.additionalHeadless }}-headless{{ end }}.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}
+            {{- end }}
+            {{- if and .Values.receive.enabled $query.dnsDiscovery.enabled }}
+            - --store=dnssrv+_grpc._tcp.{{ include "common.names.fullname" . }}-receive{{ if .Values.receive.service.additionalHeadless }}-headless{{ end }}.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}
             {{- end }}
```
Comment (carrodher) [inline bitnami/thanos/templates/query/deployment.yaml:90]: This block was implemented (using the previous approach) in a different PR a couple of hours ago, see https://github.com/bitnami/charts/pull/5467.

Can you rebase the changes from the master branch? Thanks

**Comment by Cellebyte** [comment]
messed up.

File: bitnami/thanos/templates/query/deployment.yaml (hunk 3 above)
Comment (Cellebyte) [reply bitnami/thanos/templates/query/deployment.yaml:90]: rebased it in  #5475 @carrodher sorry for breaking it here ^^.

**Pull request closed without merging**